test/src/tests-serverModel.ts

```diff
@@ -0,0 +1,1675 @@
+                                Brand: 'mParticle',
+                                Variant: 'blue',
+                                Category: 'product',
+                                Position: 1,
+                                CouponCode: 'FIDDY',
+                                TotalAmount: '41.50',
+                                Attributes: {
+                                    foo: 'bar',
+                                    fizz: 'bizz',
+                                },
+                            },
+                            {
+                                Sku: 'SKU-67890',
+                                Name: 'Some Other Item',
+                                Price: '37',
+                                Quantity: '5',
+                                Brand: 'mParticle',
+                                Variant: 'red',
+                                Category: 'not-product',
+                                Position: 2,
+                                CouponCode: 'FIDDY',
+                                TotalAmount: '36.50',
+                                Attributes: {
+                                    fizz: 'buzz',
+                                },
+                            },
+                        ],
+                    },
+                ],
+            } as unknown) as IUploadObject;
+
+            const expectedProducts = [
+                {
+                    id: 'SKU-12345',
+                    nm: 'Some Item',
+                    pr: 42,
+                    qt: 3,
+                    br: 'mParticle',
+                    va: 'blue',
+                    ca: 'product',
+                    ps: 1,
+                    cc: 'FIDDY',
+                    tpa: 41.5,
+                    attrs: {
+                        foo: 'bar',
+                        fizz: 'bizz',
+                    },
+                },
+                {
+                    id: 'SKU-67890',
+                    nm: 'Some Other Item',
+                    pr: 37,
+                    qt: 5,
+                    br: 'mParticle',
+                    va: 'red',
+                    ca: 'not-product',
+                    ps: 2,
+                    cc: 'FIDDY',
+                    tpa: 36.5,
+                    attrs: {
+                        fizz: 'buzz',
+                    },
+                },
+            ];
+            const actualDTO = ServerModel.convertEventToDTO(uploadObject);
+
+            expect(actualDTO.cu).to.equal('USD');
+            expect(actualDTO.pi[0].pil, 'ProductImpressionList').to.equal(
+                'test-product-impression'
+            );
+            expect(actualDTO.pi[0].pl, 'ProductList').to.eql(expectedProducts);
+        });
+
+        it('should add profile to DTO', () => {
+            const uploadObject = ({
+                EventDataType: Types.MessageType.Profile,
+                ProfileMessageType: 'foo-message-type',
```

Comment:
I didn't recall what `ProfileMessageType` was, but after searching, it [looks like it should be a number](https://github.com/mParticle/mparticle-web-sdk/blob/b3ac3630e998e8e192c3d6db5fe47c01de909fa3/src/serverModel.ts#L148), but by setting this `as unknown` below it is not typing it properly.  By assigning it this way, 
```const uploadObject: IUploadObject = ({ ```
you'd have to ensure all the properties existed on this object, so I'm not sure what the best approach is.

Reply:
The reason for setting this to `unknown` and then to `IUploadObject` is so that our tests can have smaller, more basic sample events without getting bogged down with unnecessary requirements. The actual assignment of `pet` inside of `convertEventToDTO` is actually testing the validation.